File: src/mobile/lib/app/utils/dto/produtos/produto_detail_dto.dart
class ProdutoDetailDto {
  ProdutoDetailDto({
    required this.id,
    required this.nome,
    required this.descricao,
    required this.imageUrl,
    required this.preco,
    required this.unidadeMedida,
    required this.estoque,
    required this.rating,
    required this.ratingCount,
    required this.isAtivo,
    required this.isFavorito,
  });
  late final String id;
  late final String nome;
  late final String descricao;
  late final String imageUrl;
  late final double preco;
  late final String unidadeMedida;
  late final String codigoBarras;
  late final int estoque;
  late final num rating;
  late final int ratingCount;
  late final bool isAtivo;
  late final bool isFavorito;

  ProdutoDetailDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    descricao = json['descricao'];
    imageUrl = json['imageUrl'];
    preco = json['preco'];
    unidadeMedida = json['unidadeMedida'];
    codigoBarras = json['codigoBarras'];
    estoque = json['estoque'];
    rating = json['rating'];
    ratingCount = json['ratingCount'];
    isAtivo = json['isAtivo'];
    isFavorito = json['isFavorito'];
  }

  bool isDisponivel() => isAtivo && estoque > 0;

  String getDisponiveis() => !isDisponivel()
      ? 'Indisponível'
      : estoque == 1
          ? '1 disponível'
          : '$estoque disponíveis';
}

File: src/services/Catalogo/Catalogo.API/Data/Dto/ProdutoDetailDto.cs
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json;

namespace Catalogo.API.Data.Dto
{
  public record ProdutoDetailDto
  {
    [BsonId]
    public string Id { get; set; } = null!;

    public string ImageUrl { get; set; } = null!;

    public string Nome { get; set; } = null!;

    public string Descricao { get; set; } = null!;

    public decimal Preco { get; set; }

    public string UnidadeMedida { get; set; } = null!;

    public string CodigoBarras { get; set; } = null!;

    public double Rating { get; set; }

    public int RatingCount { get; set; }

    public int Estoque { get; set; }

    public bool IsAtivo { get; set; }

    public bool IsFavorito { get; set; }

    public override string ToString()
    {
      return JsonSerializer.Serialize(this);
    }
  }
}
